implemented recursive version of check_downstream

maximilianmordig opened the pull request.

maximilianmordig commented:
This addresses issue #7

waddella commented:
I have an issue when I select install all dependencies and then click the compute graph button:

![image](https://user-images.githubusercontent.com/7786462/123612756-214b0c00-d803-11eb-9d6b-b599e132a654.png)

maximilianmordig commented:
There was a hard-coded dependency to this project dir. I fixed this.

waddella reviewed (commented):
some comments above

maximilianmordig commented on R/dependencies.R:
```diff
@@ -0,0 +1,623 @@
+#' \dontrun{
+#' dependency_table()
+#' }
+dependency_table <- function(project = ".", feature = NULL,
```
fixed

maximilianmordig commented:
I did the following changes in the last commit:
replaced testthat::test_dir by devtools::test because they don't seem to agree (probably because namespace not loaded); fixed error when local_repos was NULL; added project to local_repos for shiny app; improved shiny app layout; run installation after shiny app closes as a job; compute dependency graph initially in the shiny app; put branch below repo label in graph; all addins now run as rstudio jobs rather than in the main session

waddella commented:
thanks @maximilianmordig, I will test it and provide feedback

waddella commented:
@maximilianmordig I will merge this PR and create new delta issues.
